### PR TITLE
added dropdowns in navigation bar

### LINKED ISSUE #277: Add menus to title bar
There are so many buttons in the title bar, which do not fit on small screens and also creates a lot of noise. 

Lets categorize the options and create different menus for them.
```
1 Run Button
2 Language Selector
3 Input
4 File
   4.1 New
   4.2 Save
   4.3 Upload
   4.4 Download
5 Settings
6 Share
7 Shortcuts
8 Username
  8.1 saved codes
  8.2 logout
```

### PR DESCRIPTION
**Pull request by rational-kunal**
Fixed #277, added drop downs for file option and user option.

Screenshot from 2020-05-16 09-00-52
Screenshot from 2020-05-16 09-00-53
Screenshot from 2020-05-16 09-00-55

### CODE REVIEW

**Comment by rational-kunal** [comment]
Any comment @hereisnaman

**Comment by rational-kunal** [comment]
Any comments @hereisnaman

**Comment by hereisnaman** [comment]
No need to nest the user profile inside the dropdown. The dropdown toggle should be the user name, linking to the profile.

**Comment by rational-kunal** [comment]
Is this the structure you are proposing

<username>
- my profile (this will link to profile)
- saved codes
- logout

**Comment by rational-kunal** [comment]
Is this the structure you are proposing

username
--> my profile (this will link to profile)
--> saved codes
--> logout

**Comment by hereisnaman** [comment]
> Is this the structure you are proposing
> 
> username
> --> my profile (this will link to profile)
> --> saved codes
> --> logout

Seems like the dropdown does not open on hover. Let it be that way only, got head with this:

username
--> profile (this will link to profile)
--> saved codes
--> logout

**Comment by rational-kunal** [comment]
Done @hereisnaman

File: src/components/editor/MenuBar.vue [hunk 1]
```diff
@@ -11,27 +11,43 @@
...
-            </button>
-            <input type="file" ref="fileUpload" style="display:none" @change="uploadCode">
-            <button type="button" id="uploadFile" class=" btn btn-sm btn-menu" @click="selectFile">
-              Upload <span class="fa fa-folder-open" aria-hidden="true"></span>
-            </button>
+
+            <div class="btn-group" :class="{ fileOptionOpen : isFileOptionOpen}"  @click="fileOptionOpen">
+              <button id="panelLang" type="button" class="btn btn-sm btn-menu"
+                      aria-haspopup="true" aria-expanded="false" @blur="fileOptionClose" >
+                file options<span class="fa fa-caret-down"></span>
```
Comment (hereisnaman) [inline src/components/editor/MenuBar.vue:24]: it should be 'file' only. Remove 'options'.

Also add a file specific icon.

**Comment by hereisnaman** [comment]
Please add updated and fullscreen screenshots.

**Comment by rational-kunal** [comment]
@hereisnaman I have applied requested changes, for icon placement to maintain consistency I have placed file icon to the right side, just like language selection dropdown

Here are screenshots

Screenshot from 2020-05-16 22-22-58

Screenshot from 2020-05-16 22-22-55

Screenshot from 2020-05-16 22-22-52

**Comment by rational-kunal** [comment]
done @hereisnaman

**Comment by hereisnaman** [comment]
@rational-kunal Good Work.

On a further request can you do the following:
- Create a new menu after File, name it 'View'. Put Settings and Shortcuts inside that menu.
- Have the File icon, same as the ones used with New.

**Comment by hereisnaman** [comment]
@rational-kunal can you resolve the conflicts and add new screen shots

**Comment by rational-kunal** [comment]
Updated screen shots

Screenshot from 2020-05-17 21-37-41

Screenshot from 2020-05-17 21-37-44

Screenshot from 2020-05-17 21-37-45

Screenshot from 2020-05-17 21-37-49

**Comment by hereisnaman** [comment]
Add an icon with the 'View' and I'll merge this.

**Comment by rational-kunal** [comment]
updated screenshots with view icon

Screenshot from 2020-05-17 22-45-43

Screenshot from 2020-05-17 22-45-49

Screenshot from 2020-05-17 22-45-52

Screenshot from 2020-05-17 22-45-57

**Pull request merged**